frontend/appflowy_flutter/lib/plugins/database_view/widgets/row/cells/checklist_cell/checklist_progress_bar.dart

```diff
@@ -18,13 +25,41 @@ class _ChecklistProgressBarState extends State<ChecklistProgressBar> {
     return Row(
       children: [
         Expanded(
-          child: LinearPercentIndicator(
-            lineHeight: 4.0,
-            percent: widget.percent,
-            padding: EdgeInsets.zero,
-            progressColor: Theme.of(context).colorScheme.primary,
-            backgroundColor: AFThemeExtension.of(context).progressBarBGColor,
-            barRadius: const Radius.circular(5),
+          child: Row(
+            children: [
+              if (widget.tasks.isNotEmpty &&
+                  widget.tasks.length <= widget.segmentLimit) ...[
+                for (int i = 0,
+                        j = widget.tasks.where((e) => e.isSelected).length;
+                    i < widget.tasks.length;
```

Comment:
I think you can calculate what is the number of completed tasks and incompleted tasks, then do a spread operator on a List.generate